php-frontend/src/main/java/org/sonar/php/filters/SuppressWarningFilter.java

```diff
@@ -118,7 +117,7 @@ private static Range<Integer> computeLineRange(PHPTree phpTree, SyntaxToken star
       .map(PHPTree::getLastToken)
       .map(SyntaxToken::endLine)
       .orElse(endToken.endLine());
-    return Range.closed(startLine, endLine);
+    return IntStream.rangeClosed(startLine, endLine).boxed().collect(Collectors.toSet());
```

Comment:
I think you're right, I did not think about suppressing whole files just simple lines. I will try to find a better solution.

Reply:
As we believe this is not a super urgent improvement and will require some time to implement, this will be handled in https://sonarsource.atlassian.net/browse/SONARPHP-1452